Sprint 14 Домашнее задание

AlexanderKorytin opened the pull request.

AlexanderKorytin committed:
Спринт14
 1. Написана функция fun searchDebounce для поиска музыки по вводимому запросу с ожиданием 1 сек.
 2. Код вызова fun searchDebounce вынесен в отдельный поток для реализации ожидания 0.5 сек, для недопущения ошибки отправки запроса с путым полем ввода при быстром стирании текста до пустого значения

AlexanderKorytin committed:
Спринт14
1. Поток в onTextChanged удален(ошибочное отображение плейсхолдера все равно ппоисходило: при трех пробелах, после стирания первого - небольшая пауза, далее очень быстрое стирание)
2. в fun getMusic внесено условие проверки на пустой текс заппоса

AlexanderKorytin committed:
Спринт14
1.поправлено одновременное отображение плейсхолдера или результатлв поиска с историей поиска при быстром наборе и стирании запроса
2. Кликабельность кнопки очистить поле ввода отключается на время обработки запроса

AlexanderKorytin committed:
Спринт14
1.поправлено одновременное отображение плейсхолдера или результатлв поиска с историей поиска при быстром наборе и стирании запроса
2. Кликабельность кнопки очистить поле ввода отключается на время обработки запроса
3. Сверстан ProgressBar и написан код установки его видимости

AlexanderKorytin committed:
Спринт14
1. FindActivit - инициализация View переписана на binding

AlexanderKorytin committed:
Спринт14
1. Добавлен debounce для обработки нажатия на элемент списка результатов поиска

AlexanderKorytin committed:
Спринт14
1. Progress bar приведен к соответствию верстки
2. видимость progressBar устанавливается на true с момента выполнения запроса, как сказано в задании
3. Вресурсы добавлена иконка кнопки Pause для светлой и темной тем

AlexanderKorytin committed:
Спринт14
1. Вресурсы добавлена иконка кнопки Pause для светлой и темной тем
2. Написан медиа плеер, таймер пока считает от 00:00

AlexanderKorytin committed:
Спринт14
 1. Таймер плеера считает от 00:30
 2. Добавлена вибрация на нажатие кнопки играть MediaActivity
 3. Как временная заглушка в MainActivity в buttonMedia.setOnClickListenerWithViber написана логика передачи последнего прослушанного трека в плеер

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/FindViewHolder.kt:
```diff
@@ -9,6 +9,7 @@ import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.resource.bitmap.RoundedCorners
+import kotlinx.coroutines.withContext
```
🍏 Остались неиспользуемые импорты. Для форматирования кода и удаления неиспользуемых импортов применяй следующие сочетания клавиш:
Windows: Control+Alt+L
MacOS: Command+Option+L
Больше горячих клавиш можно найти здесь: https://developer.android.com/studio/intro/keyboard-shortcuts

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/FindActivity.kt:
```diff
@@ -32,20 +36,17 @@ const val TRACK_HISTORY_SHAREDPREFERENCES = "Track history"
-    private lateinit var clearSearchHistory: Button
 
+    private lateinit var bindingFindActivity: ActivityFindBinding
+    private var isClicedTrackAllowed = true
```
🍏 
```suggestion
    private var isClickTrackAllowed = true
```

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/FindActivity.kt:
```diff
@@ -73,18 +74,10 @@ class FindActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_find)
+        bindingFindActivity = ActivityFindBinding.inflate(layoutInflater)
```
👏 Круто, что используешь ViewBinding!

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/FindActivity.kt:
```diff
@@ -73,18 +74,10 @@ class FindActivity : AppCompatActivity() {
     @RequiresApi(Build.VERSION_CODES.R)
```
🍏 У тебя в приложении заявлено, что minSdk=29, targetSdk=33. При этом в этой строке указано, что метод onCreate сработает только для устройств с установленной системой андроид, соответствующей 30 Sdk. Что это значит на практике? Если бы у тебя действительно ниже был код, который работает с 30 Sdk, то приложение упало бы при попытке открытия экрана на 29 sdk.
Твой код должен работать на всех Sdk от заявленной minSdk по targetSdk. Аннотацию RequiresApi вообще лучше не использовать в приложении, она подходит для библиотек.

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -22,7 +28,13 @@ class MainActivity : AppCompatActivity() {
         }
 
         buttonMedia.setOnClickListenerWithViber {
+            val trackListJson = sharedPreferencesMainActivity.getString(SEARCH_HISTORY_TRACK_LIST, null)
```
🍏 Интересное решение, чтобы не показывать пустой экран) Вообще с этой кнопки будет переход на отдельный экран работы с избранным и плейлистами. Поэтому я рекомендую уже сейчас не смешивать activity с плеером и активити медиатеки

kgornostaeva commented:
🍏 Крестик в поле поиска отличается от макетов:
<img width="240" alt="Снимок экрана 2023-08-11 в 22 13 51" src="https://github.com/AlexanderKorytin/PlaylistMaker/assets/48408134/e1003f33-fcaa-4aab-b845-2c4fbfb1ea75">

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/MediaActivity.kt:
```diff
@@ -44,27 +59,43 @@ class MediaActivity : AppCompatActivity() {
         roundedCorners = (dpToPx(widthDisplay.toFloat(), this)) / cornersRatio
```
🍏 Можно использовать стандартные инструменты. R.dimen.album_cover_corner_radius - сохраненное в ресурсах значение радиуса в dp:
```suggestion
        roundedCorners = resources.getDimensionPixelSize(R.dimen.album_cover_corner_radius)
```

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/PlayerState.kt:
```diff
@@ -0,0 +1,10 @@
+package com.example.playlistmaker
+
+enum class PlayerState(value: Int) {
```
🍏 value не используется, можно удалить. В kotlin значения enum классов могут обходиться без параметров в конструкторе

AlexanderKorytin committed:
Спринт14
1. изменена кнопка очистки опля ввода FindActivity в соответсвии с проектом
2. создана MediaLibraryActivity с кнопкой back
3. в MainActivity при нажатии buttonMedia открывается MediaLibraryActivity, старая реализация с открытием плеера удалена
4. Время воспроизведения трека идет о 00:00 вверх

AlexanderKorytin committed:
Спринт14
1. изменена кнопка очистки опля ввода FindActivity в соответсвии с проектом
2. создана MediaLibraryActivity с кнопкой back
3. в MainActivity при нажатии buttonMedia открывается MediaLibraryActivity, старая реализация с открытием плеера удалена
4. Время воспроизведения трека идет о 00:00 вверх

AlexanderKorytin committed:
Спринт14
1. отступ кнопки очистить поле ввода от конца поля установлен 12dp
2.  private var isClicedTrackAllowed изменено название на private var isClickTrackAllowed
3. изменен конструктор enum class PlayerState

AlexanderKorytin committed:
Спринт14
1. в SettingsActivity в вызовы неявных Intent добавлены блоки try{} catch{}

AlexanderKorytin committed:
устранены замечания и выолнены рекомендации ревью

AlexanderKorytin committed:
Спринт 14
1. добавлена вибрация в кнопку назад MediaLibraryActivity

kgornostaeva commented on .idea/deploymentTargetDropDown.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
🍏 Добавь всю папку /.idea в файл  .gitignore, чтобы она не подгружалась в Pull request

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/MediaLibraryActivity.kt:
```diff
@@ -0,0 +1,22 @@
+        val view = bindingMediaLibraryActivity.root
+        setContentView(view)
+
+        bindingMediaLibraryActivity.backMediaLibray.setOnClickListenerWithViber {
```
🍏 Круто, что экспериментируешь и добавляешь особенности в свое приложение! Имей в виду, что некоторые производители встраивают в телефоны зубодробительную вибрацию, поэтому важно тестировать UX на разных моделях, чтобы понять, комфортно ли будет пользователю)

AlexanderKorytin committed:
Спринт 14
1. добавлена анимация Scale в MediaActivity на кнопку playPause
